prusti-viper/src/encoder/builtin_encoder.rs

```diff
@@ -92,6 +105,35 @@ impl BuiltinEncoder {
                 posts: vec![],
                 body: None,
             },
+            BuiltinFunctionKind::ArrayLookupPure { array_ty, array_len, return_ty } => {
+                let array_ty = if let vir::Type::TypedRef(ty) = array_ty { ty } else { unreachable!() };
+                let type_pred = format!("Array${}${}", array_len, array_ty);
+                let self_var = vir::LocalVar::new("self", vir::Type::TypedRef(type_pred.clone()));
+                let idx_var = vir_local!{ idx: Int };
+
+                vir::Function {
+                    name: fn_name,
+                    formal_args: vec![
+                        // self,
+                        self_var.clone(),
+                        // idx,
+                        idx_var.clone(),
+                    ],
+                    return_type: return_ty,
+                    pres: vec![
+                        // idx < {len}
+                        vir!([vir::Expr::Local(idx_var, vir::Position::default())]  < [vir::Expr::from(array_len)]),
```

Comment:
```suggestion
                        vir!([vir::Expr::local(idx_var)] < [array_len.into()]),
```
Does the RHS with `into()` work?

Reply:
no, it cannot infer the type.. so i think `vir::Expr::from(array_len)` is the shortest version, i mean `from` is the same as `into` internally anyway.